pyface/ui/qt4/tasks/advanced_editor_area_pane.py

```diff
@@ -304,18 +304,14 @@ def get_dock_widgets_for_bar(self, tab_bar):
     def get_dock_widgets_ordered(self, visible_only=False):
         """ Gets all dock widgets in left-to-right, top-to-bottom order.
         """
-        def compare(one, two):
-            y = cmp(one.pos().y(), two.pos().y())
-            return cmp(one.pos().x(), two.pos().x()) if y == 0 else y
-
         children = []
         for child in self.children():
             if (child.isWidgetType() and child.isVisible() and
                 ((isinstance(child, QtGui.QTabBar) and not visible_only) or
                  (isinstance(child, QtGui.QDockWidget) and
                   (visible_only or not self.tabifiedDockWidgets(child))))):
                 children.append(child)
-        children.sort(cmp=compare)
+        children.sort(key=lambda x: (x.pos().y(), x.pos().x()))
```

Comment:
One suggestion for clarity: perhaps use a different variable name for the `lambda`, since the `x` is ambiguous in context.  Eg. `lambda child: (child.pos().y(), child.pos().x())` is better.